src/main/java/net/neoforged/neoforge/common/conditions/ElementExistsCondition.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.conditions;
+
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.Holder;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+
+public record ElementExistsCondition<TRegistry>(ResourceKey<TRegistry> registryKey) implements ICondition {
```

Comment:
```suggestion
public record ElementExistsCondition<T extends Registry<?>>(ResourceKey<T> registryKey) implements ICondition {
```
does this work? Not a fan of `TSomething`.

Reply:
I could change it to `T` but having it extend `Registry` doesnt help as we are not looking for a "Registry" instance just a "registry type". That being `Item`, `Block`, `BlockEntityType`, `EntityType` etc, none of which extend `Registry`.

Reply:
Ok, no need for `extends Registry<?>` then. I was confused by the `TRegistry` naming.